charts/gateway-operator/templates/deployment.yaml

```diff
@@ -30,7 +30,25 @@ spec:
       - args:
         - --health-probe-bind-address=:8081
         - --metrics-bind-address=127.0.0.1:8080
+        {{- if .Values.featureFlags }}
+        {{- if eq .Values.featureFlags.controlPlaneControllerEnabled false }}
+        - --enable-controller-controlplane=false
+        {{- end }}
+        {{- if eq .Values.featureFlags.gatewayControllerEnabled false }}
+        - --enable-controller-gateway=false
+        {{- end }}
+        {{- if eq .Values.featureFlags.dataPlaneControllerEnabled false }}
+        - --enable-controller-dataplane=false
+        {{- end }}
+        {{- if eq .Values.featureFlags.dataPlaneBlueGreenControllerEnabled false }}
+        - --enable-controller-dataplane-bluegreen=false
+        {{- end }}
+        {{- if .Values.featureFlags.aiGatewayControllerEnabled }}
+        - --enable-controller-aigateway=true
+        {{- end }}
+        {{- end }}
```

Comment:
How about just setting this to whatever's set in the `values.yaml`?

By checking values here we keep "defaults" in 2 place: here and in `values.yaml` (not mentioning the operator's default for flags).

Reply:
Because our default values are all true,
If the user does not specify, they should all be true
Only when the user specifies false, do we need to set them as false.

Reply:
Got it. But this way we need to update flags in 2 places if defaults change.

Reply:
Yes. 

How about this? `--enable-controller-dataplane=.Values.featureFlags.dataPlaneControllerEnabled`

Reply:
This could work yes